src/snowflake/cli/api/project/schemas/updatable_model.py

```diff
@@ -14,14 +15,42 @@ def __init__(self, *args, **kwargs):
         except ValidationError as e:
             raise SchemaValidationError(e)
 
-    def update_from_dict(
-        self, update_values: Dict[str, Any]
-    ):  # this method works wrong for optional fields set to None
-        for field, value in update_values.items():  # do we even need this?
-            if getattr(self, field, None):
-                setattr(self, field, value)
+    def update_from_dict(self, update_values: Dict[str, Any]):
+        """
+        Takes a dictionary with values to override.
+        If the field type is subclass of a UpdatableModel, it's update_from_dict() method is called with
+        the value to be set.
+        If not, we use simple setattr to set new value.
+        Values provided are validated against original restrictions, so it's impossible to overwrite string field with
+        integer value etc.
+        """
+        for field, value in update_values.items():
+            if field in self.model_fields.keys():
+                if (
+                    self._is_field_an_updatable_model(field)
+                    and field in self.model_fields_set
+                ):
+                    getattr(self, field).update_from_dict(value)
+                else:
+                    setattr(self, field, value)
         return self
 
+    def _is_field_an_updatable_model(self, name: str) -> bool:
```

Comment:
This feels really brittle. Wouldn't it be sufficient to check if `update_from_dict` is defined on the getattr result, and invoke it if so? Python isn't Java, duck typing can be appropriate in the right circumstances. I do believe this is one of them.

Reply:
There are moments in junior dev's life, when one encounters a problem, to which a simplest solution fails. So one comes up with a little bit more sophisticated solution, but it also fails. So, one adds some more logic, but it still fails, but just a little bit more logic fixes it. Solution is a bit complicated, but concise, and seems very smart.
And then, one reads a comment in a review, and spends 30min with his face hidden in palms, whispering "FFS it was so simple, just needed to take a step back".
Yeah, this is such moment.
Updated.

Reply:
Lol, fair enough. Junior or not, we all have these moments. Part of why code reviews are valuable.